homeassistant/components/hue/switch.py

```diff
@@ -68,19 +84,15 @@ def async_add_entity(event_type: EventType, resource: SensingService) -> None:
 class HueSensingServiceEnabledEntity(HueBaseEntity, SwitchEntity):
     """Representation of a Switch entity from Hue SensingService."""
 
-    _attr_entity_category = EntityCategory.CONFIG
-    _attr_device_class = SwitchDeviceClass.SWITCH
+    controller: ControllerType
+    resource: SensingServiceResource
 
-    def __init__(
-        self,
-        bridge: HueBridge,
-        controller: LightLevelController | MotionController,
-        resource: SensingService,
-    ) -> None:
-        """Initialize the entity."""
-        super().__init__(bridge, controller, resource)
-        self.resource = resource
-        self.controller = controller
+    entity_description = SwitchEntityDescription(
+        key="sensing_service_enabled",
```

Comment:
so this sensor itself is never created? Can't we just make entity_description an attribute

Reply:
I'm not sure what you mean here, its already an attribute

Reply:
I mean, pass in the entity description in the constructor

Reply:
You are aware that this subclass doesnt have a constructor and its parent is the HueBaseEntity ?

Reply:
I like it to be consequent with all others where the entity description is a class attribute (at least for all the static descriptions)

Reply:
I mean it had a constructor until you removed it 😂

But yea I remember the other hue entities having the same setup, so hmm.

I would advice to put `_attr_has_entity_name` in the superclass tho

Reply:
I don't agree :-)
I'd rather not use the attributes and entity descriptions mixed up, its one or the other for clarity/readability.

So, "_attr_has_entity_name" is only set whenever an entity does not use the entity description, which I believe is only true for the scene entities.

Reply:
Right, I always set it so all new entities would have that property automatically as well, but you explanation is fine :)